openedx/features/discounts/applicability.py

```diff
@@ -82,6 +82,30 @@ def get_discount_expiration_date(user, course):
     return discount_expiration_date
 
 
+def AA759_can_show_coupon(user, course):
+    """
+    Check whether this combination of user and course
+    can receive the AA-759 experiment discount.
+    """
+    # Course end date needs to be in the future
+    if course.has_ended():
+        return False
```

Comment:
Do we instead want to check upgrade deadline?

I think this call and the `'verified'` mode check below could both be replaced by the more complicated `can_show_verified_upgrade` used by similar code.

Reply:
that depends on the dynamic upgrade deadline which is not what we want here.  
the verified mode check does check if the course upgrade deadline has expired

Reply:
i've pulled out the code we can use within can_show_verified_ugprade into is_mode_upsellable.  
since the existing code was not tested i should probably write a test for is_mode_upsellable so i'll be working on that now